buildSrc/src/main/java/org/elasticsearch/gradle/Version.java

```diff
@@ -16,35 +16,43 @@
     /**
      * Suffix on the version name.
      */
-    private final String suffix;
+    private final String qualifier;
 
     private static final Pattern pattern =
             Pattern.compile("(\\d)+\\.(\\d+)\\.(\\d+)(-alpha\\d+|-beta\\d+|-rc\\d+)?(-SNAPSHOT)?");
 
-    public Version(int major, int minor, int revision, String suffix, boolean snapshot) {
+    Version(int major, int minor, int revision, String qualifier, boolean snapshot) {
         Objects.requireNonNull(major, "major version can't be null");
         Objects.requireNonNull(minor, "minor version can't be null");
         Objects.requireNonNull(revision, "revision version can't be null");
         this.major = major;
         this.minor = minor;
         this.revision = revision;
         this.snapshot = snapshot;
-        this.suffix = suffix == null ? "" : suffix;
+        if (qualifier == null || qualifier.isEmpty()) {
+            this.qualifier = "";
+        } else if (qualifier.startsWith("-")) {
+            this.qualifier = qualifier;
+        } else {
+            this.qualifier = "-" + qualifier;
```

Comment:
Do we need to do the formatting here (prefixing with dash), or could this be done when printed?

Reply:
We used to store it with a dash so existing code   expected it, but I didn't want to add have it in the property, so this takes care of that. I think the cleanest way would be to have an enum for the qualifier.

Reply:
An enum won't work since these are numbered (or it would require separate variable to keep track of the number). But I don't see why the leniency needs to exist here to add the dash if it doesn't exist. I think the qualifier should always have no dash internally, just like we don't have dot prefixes on the numeric parts of the version.